data/cdc2024nhphrn/cdc2024nhphrn-extraction.md

```diff
@@ -0,0 +1,136 @@
+
+
+First, we `import` python modules needed:
+
+```python
+import pandas as pd
+import yaml
+from shedding_hub import folded_str
+```
+
+
+```python
+#load the data;
+test = pd.read_excel('NHPHRN INHERENT Public Use Files and Data Dictionary.xlsx', sheet_name='Case list test results')
+
+columns_to_keep = [
+    'nh', 'outbreak_id', 'study_id', 'type', 'event', 
+    'colldate', 'day', 'day_v2', 'antiviral_start', 'antigen','pcr', 'NHConductedPCRTest', 'Ct', 
+    'VL', 'logVL', 'VL_status', 'variant', 'lineage'
+]
+test = test[columns_to_keep]
+
+descriptive = pd.read_excel('NHPHRN INHERENT Public Use Files and Data Dictionary.xlsx', sheet_name='Case descriptive information')
+
+columns_to_keep = [
+    'study_id', 'enr_date', 'casedate', 
+    'latesdat', 'symptom', 'age', 'gender', 'bio_sex', 
+    'race', 'ethnicity', 'paxlovid', 'remdesivir', 
+    'molnupiravir', 'monoclonal', 'steroid', 'vaccine'
+]
+
+descriptive = descriptive[columns_to_keep]
+
+merged_df = pd.merge(test, descriptive, on='study_id', how='left')
+```
+
+Finally, the data is formatted and output as a YAML file.
+
+
+```python 
+participants = []
+
+for patient_id, group in merged_df.groupby('study_id'):
+    participant = {'attributes': {}, 'measurements': []}
+
+    if pd.notna(group['age'].iloc[0]):
+        participant['attributes']['age'] = int(group['age'].iloc[0])
+
+    vaccine_val = group['vaccine'].iloc[0]
+    if pd.isna(vaccine_val) or vaccine_val == 'COVID-19 vaccination status unknown':
+        pass
+    else:
+        vaccine_numeric = pd.to_numeric(vaccine_val, errors='coerce')
+        if pd.notna(vaccine_numeric):
+            participant['attributes']['vaccinated'] = vaccine_numeric > 0
+
+    if pd.notna(group['bio_sex'].iloc[0]):
+        sex = ('female' if group['bio_sex'].iloc[0] == 'Female at birth'
+            else 'male' if group['bio_sex'].iloc[0] == 'Male at birth'
+            else 'unknown')
+        if sex != 'unknown':
+            participant['attributes']['sex'] = sex
+
+    if pd.notna(group['ethnicity'].iloc[0]):
+        ethnicity = group['ethnicity'].iloc[0]
+        ethnicity_val = ('not hispanic' if ethnicity == 'Non-White'
+                        else 'hispanic' if ethnicity == 'Hispanic'
+                        else 'unknown')
+        if ethnicity_val != 'unknown':
+            participant['attributes']['ethnicity'] = ethnicity_val
+
+    if pd.notna(group['race'].iloc[0]):
+        race = group['race'].iloc[0]
+        race_val = ('white' if race == 'White'
+                    else 'black' if race == 'Black or African American'
+                    else 'other' if race in ['Other', 'American Indian or Alaska Native', 'Native Hawaiian or Other Pacific Islander']
+                    else 'unknown')
+        if race_val != 'unknown':
+            participant['attributes']['race'] = race_val
```

Comment:
Same question regarding `unknown` for race.

https://github.com/shedding-hub/shedding-hub/blob/66fd041bf99ccf2e7ed3f24b03f31dc5c2428452/data/.schema.yaml#L162-L170